aopy/analysis/base.py

```diff
@@ -1810,4 +1811,95 @@ def calc_confidence_interval_overlap(CI1, CI2):
+
+    Notes:
+        The function applies a Gaussian smoothing filter to the data if `smooth_timeseries=True`. The smoothing width is 
+        provided in milliseconds and will be converted to samples based on the sample rate.
+        
+        If `nfolds < 2`, no cross-validation is performed, and the function will calculate the accuracy based on the entire dataset.
+        If `nfolds >= 2`, k-fold cross-validation is performed, and decoding accuracy is calculated for each fold.
+    """
+
+    ntime, nch, ntrials = data.shape 
+    nlabels = len(np.unique(labels))
+
+    # Perform n-fold xval LDA at each time point 
+    decoding_accuracy = np.zeros((ntime-lags, nfolds))*np.nan
+    weights = np.zeros((ntime-lags, nlabels, (1+lags)*nch, nfolds))*np.nan # (ntime, ntargets, nfeatures, nfolds)
+    cm = np.zeros((ntime-lags, nlabels, nlabels, nfolds))*np.nan
+    nwind = data.shape[0] - lags
+    for iwind in range(nwind):
+        lda = LinearDiscriminantAnalysis(solver='eigen', shrinkage=regularization)
+        end_wind = iwind+lags+1
+
+        # If there are nans at this timestep, return nan in all places
+        if np.sum(np.isnan(data[iwind:end_wind,:,:])):
+            if nfolds < 2:
+                decoding_accuracy[iwind] = np.nan
+                weights[iwind,:,:] = np.nan
+                cm[iwind,:,:] = np.nan
+            else:
+                decoding_accuracy[iwind,:] = np.nan
+                if return_weights:
+                    weights[iwind,:,:,:] = np.nan
+                if return_confusion_matrix:
+                    cm[iwind,:,:,:] = np.nan
+
+        else:
+            # If not cross validated
+            if nfolds < 2:
+                lda.fit(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0), labels)
+                decoding_accuracy[iwind] = lda.score(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0), labels)
+                predicted_labels = lda.predict(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0))
+                cm[iwind,:,:,0] = confusion_matrix(labels, predicted_labels)
+                                                        
+            # if cross validated: 
+            else:
+                kf = model_selection.KFold(n_splits=nfolds,shuffle=True,random_state=None)
+                for ifold, (train_idx, test_idx) in enumerate(kf.split(np.vstack(data[iwind:end_wind,:,:]).T)):
+                    # print(ifold, train_idx, test_idx, iwind, end_wind)
+                    lda.fit(np.vstack(data[iwind:end_wind,:,train_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,train_idx]).T, axis=0), labels[train_idx])
+                    decoding_accuracy[iwind, ifold] = lda.score(np.vstack(data[iwind:end_wind,:,test_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,test_idx]).T, axis=0), labels[test_idx])
```

Comment:
You need to use the mean of training data in centering testing data, in my understanding.
It means `np.vstack(data[iwind:end_wind,:,test_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,train_idx]).T`